### PR TITLE
Route block, transaction, and log queries through SQD Portal

### PR DESCRIPTION
**Pull request by iankressin**
## Summary

- route all structurally serviceable block, block-range transaction, and block-range log projections through SQD Portal, including `GET *`
- add complete field decoding, `latest`/`earliest` resolution, all-block range retrieval, safe multi-page continuation, and Ethereum parity normalization
- separate internal transaction filter fields from user projections across both Portal and RPC paths, retaining legitimate null-only rows
- hash `EventSignature` into `topic0`, reject ambiguous `EventSignature` plus `Topic0`, and preserve RPC-only query shapes
- remove Fantom completely from the chain model, parser, chain-ID conversion, bundled configuration, and supported-network documentation

## Why

The resolver field allowlists had drifted behind the EQL field enums, so wildcard and newly supported projections fell back to RPC even when Portal could serve them. The existing Portal path also lacked the metadata and query flags needed for complete block ranges and reliable pagination. Fantom support was removed because SQD no longer provides the chain dataset and the product decision is to reject the chain entirely rather than retain an RPC fallback.

## Impact

Portal is now selected by query shape rather than projected-field coverage. Explicit RPC URLs, transaction-by-hash queries, unsupported block tags, block-hash log filters, account queries, and dataset-less supported chains still use RPC. Fantom selectors and chain ID `250` are now invalid. Portal range queries continue to expose `authorization_list` as null because SQD does not provide that field.

## Validation

- `cargo build`
- `cargo test -p eql_core` — 65 passed, 0 failed; doc-tests passed
- focused pagination, projection/filter, parity, range, EventSignature, and Fantom regressions
- independent task review and final whole-branch review with no remaining Critical, Important, or Minor findings
- `git diff --check`, JSON/config validation, Fantom production-surface checks, and clean worktree
- Clippy and workspace rustfmt reproduce only the explicitly accepted pre-existing baseline; this branch adds no new diagnostic or formatting hunk

## Summary by CodeRabbit

* **New Features**
  * Expanded Portal-based querying for blocks, transactions, and logs with broader field support and improved handling of ranges/pagination.
  * Added event-signature support by translating signatures to `topic0` for log queries, with better projection behavior.
  * Improved Portal routing, including support for custom Portal base URLs.
* **Bug Fixes**
  * Prevented crashes for nullable transaction-filter results.
  * Improved decoding/projection for block and log metadata (including `block_hash` and `removed`) and consistent `authorization_list` handling.
* **Changes**
  * Removed Fantom from supported chains, parser acceptance, and default configuration; updated installation docs accordingly.
* **Chores**
  * Updated Git ignore rules to avoid tracking local worktrees.

### CODE REVIEW

File: crates/core/src/interpreter/backend/resolve_portal.rs [hunk 1]
```diff
@@ -1,21 +1,65 @@
...
+        .filter_map(|block| {
+            block
+                .get("header")
+                .and_then(|header| header.get("number"))
+                .and_then(value_to_u64)
+        })
+        .max()
+        .ok_or_else(|| {
+            anyhow::anyhow!("Portal returned a nonempty page without a usable header.number")
+        })?;
```
[the remark this answers is not in the document]
Comment (iankressin) [reply crates/core/src/interpreter/backend/resolve_portal.rs]: Fixed: next_portal_page_start now errors on any page item lacking a usable header.number instead of skipping it, so malformed items can no longer be appended while being invisible to pagination. Added a mixed valid/invalid page regression test. (branch head: c9dbc01)

File: crates/core/src/interpreter/backend/resolve_portal.rs [hunk 2]
```diff
@@ -123,13 +164,322 @@ pub fn value_to_bytes(v: &Value) -> Option<Bytes> {
...
+    }
+}
+
+/// Resolve and validate a Portal block range after tags become concrete numbers.
+pub async fn resolve_portal_range(dataset: &str, range: &BlockRange) -> Result<(u64, u64)> {
+    let start = resolve_portal_bound(dataset, &range.start()).await?;
+    let end = match range.end() {
+        Some(end) => resolve_portal_bound(dataset, &end).await?,
+        None => start,
+    };
```
[the remark this answers is not in the document]
Comment (iankressin) [reply crates/core/src/interpreter/backend/resolve_portal.rs:322]: Fixed: resolve_portal_range now fetches /head at most once per range and resolves both bounds against that single snapshot, so latest:latest can no longer straddle two consecutive heads. Covered by a mock-Portal test asserting exactly one /head request. (branch head: c9dbc01)

File: docs/superpowers/plans/2026-07-17-portal-primary-routing.md [hunk 3]
```diff
@@ -0,0 +1,930 @@
+# Portal-Primary Block/Transaction/Log Routing — Implementation Plan
+
+> **For agentic workers:** REQUIRED SUB-SKILL: Use superpowers:subagent-driven-development (recommended) or superpowers:executing-plans to implement this plan task-by-task. Steps use checkbox (`- [ ]`) syntax for tracking.
+
+**Goal:** Route all block/transaction/log field selections (including `GET *`) through SQD Portal, leaving RPC only for cases Portal structurally cannot serve.
+
+**Architecture:** Complete the per-resolver Portal field mappings + parse-back so every EQL field is Portal-serviceable (some via correct defaults), then delete the field-coverage routing gate — routing becomes purely query-shape based. Add shared Portal primitives (bloom decode, `/head`, tag→number resolution) so `latest`/`earliest` resolve via Portal. Fix Fantom's dead dataset mapping.
```
[the remark this answers is not in the document]
Comment (iankressin) [reply docs/superpowers/plans/2026-07-17-portal-primary-routing.md]: Fixed: Task 2 of the plan now carries a SUPERSEDED banner linking to the Fantom-removal plan/design, and the remaining Fantom references (architecture line, RPC-fallback constraint, self-review note) are annotated to match the full removal. (branch head: c9dbc01)

File: docs/superpowers/specs/2026-07-17-portal-primary-routing-design.md [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+|---|---|---|---|
+| EffectiveGasPrice | effectiveGasPrice | `u128` | `value_to_u128` |
+| V | v | `bool` | parity decode — see note |
+| R | r | `U256` | `value_to_u256` |
+| S | s | `U256` | `value_to_u256` |
+| MaxFeePerBlobGas | maxFeePerBlobGas | `u128` | `value_to_u128` |
+| YParity | yParity | `bool` | parity decode — see note |
+| **AuthorizationList** | *(none)* | `Option<Vec<SignedAuthorization>>` | left `None` |
+
+Implementation note: `v` and `y_parity` are both `Option<bool>` in EQL (parity semantics — the RPC path derives both from `signature().v()`). Decode both from Portal's parity value (`v`/`yParity`). **Do not reuse `value_to_status_bool` blindly** — it only handles JSON bool / integer, not hex strings, and Portal serializes numeric fields as hex strings (`"0x1"`). The plan must use/add a decoder that maps both integer `0/1` and hex `0x0`/`0x1` → `bool` (i.e. parse via `value_to_u64` first, then `!= 0`). Legacy-tx `v` values like `27/28` are an edge case; verify the actual Portal representation against a live type-0 tx and cover with a test. Confirm empirically whether Portal emits these as hex strings or JSON integers before finalizing the decoder.
```
[the remark this answers is not in the document]
Comment (iankressin) [reply docs/superpowers/specs/2026-07-17-portal-primary-routing-design.md]: Fixed: the design spec now describes value_to_status_bool as hex-aware and documents that authorization-list-only Portal queries retain null-only rows, matching the shipped contracts. (branch head: c9dbc01)

**Comment by iankressin** [comment]
Addressed the two review-body findings as well (branch head: c9dbc01):

- **Portal client timeouts (outside-diff, Major):** all Portal traffic now goes through one shared reqwest client built with connect_timeout=10s and timeout=60s (OnceLock; LazyLock is unavailable at the workspace MSRV of 1.79), so a stalled Portal endpoint can no longer hang a query indefinitely.
- **Portal routing test assertion (nitpick):** added offline unit tests for should_use_portal in resolve_transaction.rs that pin the exact e2e query shape to Portal routing, plus negative cases (by-hash, no block filter, pending bound, explicit RPC URL). Chose the focused-unit-test option over injecting a mock Portal endpoint through ExecutionEngine, which would have required a public API seam disproportionate to the nit.
